patches/minecraft/net/minecraft/item/Item.java.patch

```diff
@@ -668,13 +668,37 @@
 +        }
 +        return builder.build();
 +    }
++    
++    /**
++     * Can this Item disable a shield
++     * @param stack The ItemStack
++     * @param shield The shield in question
++     * @param entity The Entity holding the shield
++     * @param attacker The Entity holding the ItemStack
++     * @return True if this Item can disable the shield in question.
++     */
++    public boolean canDisableShield(ItemStack stack, ItemStack shield, Entity entity, Entity attacker)
++    {
++        return this instanceof ItemAxe;
++    }
++    
++    /**
++     * Is this Item a shield
++     * @param stack The ItemStack
++     * @param entity the Entity holding the ItemStack
++     * @return True if the ItemStack is considered a shield.
++     */
++    public boolean isShield(ItemStack stack, @Nullable Entity entity)
```

Comment:
The parameter can be `EntityLivingBase` type, I guess, and that will be more convenient.